protostar/commands/deploy/deploy_contract.py

```diff
@@ -0,0 +1,89 @@
+from io import TextIOWrapper
+from typing import Optional, Sequence, Union, cast
+
+from services.external_api.base_client import RetryConfig
+from starkware.starknet.cli.starknet_cli import validate_arguments
+from starkware.starknet.definitions import fields
+from starkware.starknet.public.abi_structs import identifier_manager_from_abi
+from starkware.starknet.services.api.contract_definition import ContractDefinition
+from starkware.starknet.services.api.gateway.gateway_client import GatewayClient
+from starkware.starknet.services.api.gateway.transaction import Deploy
+from starkware.starknet.utils.api_utils import cast_to_felts
+from starkware.starkware_utils.error_handling import StarkErrorCode
+from typing_extensions import TypedDict
+
+from protostar.protostar_exception import ProtostarException
+
+
+class SuccessfulGatewayResponseFacade(TypedDict):
+    code: str
+    address: str
+    transaction_hash: str
+
+
+class DeployContractException(ProtostarException):
+    pass
+
+
+async def deploy_contract(
```

Comment:
Can we name it `deploy` as the orginal?

Reply:
I wanted to find the original function in the starkware codebase and it mislead me